File: Python/Algorithms/167.py
class Solution:
    def twoSum(self, numbers, target):
        i, j = 0, len(numbers) - 1
        while i < j:
            ts = numbers[i] + numbers[j]
            if ts == target:
                break
            elif ts > target:
                j = j - 1
            else:
                i = i + 1
        return [i+1, j+1]

File: Rust/Algorithms/167.rs
impl Solution {
    pub fn two_sum(numbers: Vec<i32>, target: i32) -> Vec<i32> {
        let (mut i, mut j) = (0, numbers.len() - 1);
        // Note: *Can* use loop since we know a solution exists,
        // in general, better to break and return.
        loop {
            let ts = numbers[i] + numbers[j];
            if ts == target || i > j {
                break vec![(i + 1) as i32, (j + 1) as i32];
            }
            if ts > target {
                j -= 1;
            } else {
                i += 1;
            }
        }
    }
}
